### ISSUE TITLE
Traceback: TypeError: chmod: path should be string, bytes, os.PathLike or integer, not NoneType

### ISSUE THREAD

**Issue by swilmet** [systemd/mkosi#143]
I use mkosi git master (currently at commit a499812) on Fedora 26.

The traceback happens when running `mkosi -if`:
```
‣ Installing cache copy ...        
Traceback (most recent call last): 
  File "/usr/bin/mkosi", line 2904, in <module>                       
    main()                         
  File "/usr/bin/mkosi", line 2900, in main                           
    build_stuff(args)              
  File "/usr/bin/mkosi", line 2833, in build_stuff                    
    args.cache_pre_dev)            
  File "/usr/bin/mkosi", line 1751, in save_cache                     
    os.chmod(raw, 0o666 & ~args.original_umask)                       
TypeError: chmod: path should be string, bytes, os.PathLike or integer, not NoneType
```

My mkosi.default:
```
[Distribution]
Distribution=fedora
Release=26

[Output]
ReadOnly=yes

[Validation]
Password=something-else
```

**Comment by fthiery** [comment]
Same here (but on Arch), only seems to happen if incremental build is selected, and only if no build script is provided (e.g. mkosi.build).

This is because of the following lines:
https://github.com/systemd/mkosi/blob/master/mkosi#L2673

```
def build_image(args, workspace, run_build_script, for_cache=False):

    # If there's no build script set, there's no point in executing
    # the build script iteration. Let's quit early.
    if args.build_script is None and run_build_script:
        return None, None, None
```
This makes build_image return raw=None, in turn triggering the crash

**Comment by fthiery** [comment]
The strange part is that run_build_script should be False (since no build script is present)

**Comment by fthiery** [comment]
@swilmet could you try this please ? https://github.com/systemd/mkosi/pull/147

**Comment by poettering** [comment]
my proposal for a "fix" is waiting in #162. PTAL
